{"matches":[{"vulnerability":{"id":"GHSA-j7hp-h8jx-5ppr","dataSource":"https://github.com/advisories/GHSA-j7hp-h8jx-5ppr","namespace":"github:language:python","severity":"High","urls":[],"description":"libwebp: OOB write in BuildHuffmanTable","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"knownExploited":[{"cve":"CVE-2023-4863","vendorProject":"Google","product":"Chromium WebP","dateAdded":"2023-09-13","requiredAction":"Apply mitigations per vendor instructions or discontinue use of the product if mitigations are unavailable.","dueDate":"2023-10-04","knownRansomwareCampaignUse":"unknown","urls":["https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html?m=1","https://nvd.nist.gov/vuln/detail/CVE-2023-4863"],"cwes":["CWE-787"]}],"epss":[{"cve":"CVE-2023-4863","epss":0.93991,"percentile":0.9988,"date":"2025-06-02"}],"fix":{"versions":["10.0.1"],"state":"fixed"},"advisories":[],"risk":85.57500000000002},"relatedVulnerabilities":[{"id":"CVE-2023-4863","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-4863","namespace":"nvd:cpe","severity":"High","urls":["http://www.openwall.com/lists/oss-security/2023/09/21/4","http://www.openwall.com/lists/oss-security/2023/09/22/1","http://www.openwall.com/lists/oss-security/2023/09/22/3","http://www.openwall.com/lists/oss-security/2023/09/22/4","http://www.openwall.com/lists/oss-security/2023/09/22/5","http://www.openwall.com/lists/oss-security/2023/09/22/6","http://www.openwall.com/lists/oss-security/2023/09/22/7","http://www.openwall.com/lists/oss-security/2023/09/22/8","http://www.openwall.com/lists/oss-security/2023/09/26/1","http://www.openwall.com/lists/oss-security/2023/09/26/7","http://www.openwall.com/lists/oss-security/2023/09/28/1","http://www.openwall.com/lists/oss-security/2023/09/28/2","http://www.openwall.com/lists/oss-security/2023/09/28/4","https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/","https://blog.isosceles.com/the-webp-0day/","https://bugzilla.suse.com/show_bug.cgi?id=1215231","https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html","https://crbug.com/1479274","https://en.bandisoft.com/honeyview/history/","https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a","https://github.com/webmproject/libwebp/releases/tag/v1.3.2","https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/","https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863","https://news.ycombinator.com/item?id=37478403","https://security-tracker.debian.org/tracker/CVE-2023-4863","https://security.gentoo.org/glsa/202309-05","https://security.gentoo.org/glsa/202401-10","https://security.netapp.com/advisory/ntap-20230929-0011/","https://sethmlarson.dev/security-developer-in-residence-weekly-report-16","https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/","https://www.bentley.com/advisories/be-2023-0001/","https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/","https://www.debian.org/security/2023/dsa-5496","https://www.debian.org/security/2023/dsa-5497","https://www.debian.org/security/2023/dsa-5498","https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/","http://www.openwall.com/lists/oss-security/2023/09/21/4","http://www.openwall.com/lists/oss-security/2023/09/22/1","http://www.openwall.com/lists/oss-security/2023/09/22/3","http://www.openwall.com/lists/oss-security/2023/09/22/4","http://www.openwall.com/lists/oss-security/2023/09/22/5","http://www.openwall.com/lists/oss-security/2023/09/22/6","http://www.openwall.com/lists/oss-security/2023/09/22/7","http://www.openwall.com/lists/oss-security/2023/09/22/8","http://www.openwall.com/lists/oss-security/2023/09/26/1","http://www.openwall.com/lists/oss-security/2023/09/26/7","http://www.openwall.com/lists/oss-security/2023/09/28/1","http://www.openwall.com/lists/oss-security/2023/09/28/2","http://www.openwall.com/lists/oss-security/2023/09/28/4","https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/","https://blog.isosceles.com/the-webp-0day/","https://bugzilla.suse.com/show_bug.cgi?id=1215231","https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html","https://crbug.com/1479274","https://en.bandisoft.com/honeyview/history/","https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a","https://github.com/webmproject/libwebp/releases/tag/v1.3.2","https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/","https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863","https://news.ycombinator.com/item?id=37478403","https://security-tracker.debian.org/tracker/CVE-2023-4863","https://security.gentoo.org/glsa/202309-05","https://security.gentoo.org/glsa/202401-10","https://security.netapp.com/advisory/ntap-20230929-0011/","https://sethmlarson.dev/security-developer-in-residence-weekly-report-16","https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/","https://www.bentley.com/advisories/be-2023-0001/","https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/","https://www.debian.org/security/2023/dsa-5496","https://www.debian.org/security/2023/dsa-5497","https://www.debian.org/security/2023/dsa-5498","https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/","https://www.vicarius.io/vsociety/posts/zero-day-webp-vulnerability-cve-2023-4863"],"description":"Heap buffer overflow in libwebp in Google Chrome prior to 116.0.5845.187 and libwebp 1.3.2 allowed a remote attacker to perform an out of bounds memory write via a crafted HTML page. (Chromium security severity: Critical)","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"knownExploited":[{"cve":"CVE-2023-4863","vendorProject":"Google","product":"Chromium WebP","dateAdded":"2023-09-13","requiredAction":"Apply mitigations per vendor instructions or discontinue use of the product if mitigations are unavailable.","dueDate":"2023-10-04","knownRansomwareCampaignUse":"unknown","urls":["https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html?m=1","https://nvd.nist.gov/vuln/detail/CVE-2023-4863"],"cwes":["CWE-787"]}],"epss":[{"cve":"CVE-2023-4863","epss":0.93991,"percentile":0.9988,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<10.0.1 (python)","vulnerabilityID":"GHSA-j7hp-h8jx-5ppr"},"fix":{"suggestedVersion":"10.0.1"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-8vj2-vxx3-667w","dataSource":"https://github.com/advisories/GHSA-8vj2-vxx3-667w","namespace":"github:language:python","severity":"Critical","urls":[],"description":"Arbitrary expression injection in Pillow","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-22817","epss":0.02548,"percentile":0.84788,"date":"2025-06-02"}],"fix":{"versions":["9.0.1"],"state":"fixed"},"advisories":[],"risk":2.3951200000000004},"relatedVulnerabilities":[{"id":"CVE-2022-22817","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-22817","namespace":"nvd:cpe","severity":"Critical","urls":["https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html","https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#restrict-builtins-available-to-imagemath-eval","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security","https://security.gentoo.org/glsa/202211-10","https://www.debian.org/security/2022/dsa-5053","https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html","https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#restrict-builtins-available-to-imagemath-eval","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security","https://security.gentoo.org/glsa/202211-10","https://www.debian.org/security/2022/dsa-5053"],"description":"PIL.ImageMath.eval in Pillow before 9.0.0 allows evaluation of arbitrary expressions, such as ones that use the Python exec method. A lambda expression could also be used.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:P/I:P/A:P","metrics":{"baseScore":7.5,"exploitabilityScore":10,"impactScore":6.5},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-22817","epss":0.02548,"percentile":0.84788,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<9.0.1 (python)","vulnerabilityID":"GHSA-8vj2-vxx3-667w"},"fix":{"suggestedVersion":"9.0.1"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-3f63-hfp8-52jq","dataSource":"https://github.com/advisories/GHSA-3f63-hfp8-52jq","namespace":"github:language:python","severity":"Critical","urls":[],"description":"Arbitrary Code Execution in Pillow","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-50447","epss":0.00408,"percentile":0.6039,"date":"2025-06-02"}],"fix":{"versions":["10.2.0"],"state":"fixed"},"advisories":[],"risk":0.34884},"relatedVulnerabilities":[{"id":"CVE-2023-50447","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-50447","namespace":"nvd:cpe","severity":"High","urls":["http://www.openwall.com/lists/oss-security/2024/01/20/1","https://devhub.checkmarx.com/cve-details/CVE-2023-50447/","https://duartecsantos.github.io/2024-01-02-CVE-2023-50447/","https://github.com/python-pillow/Pillow/releases","https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html","http://www.openwall.com/lists/oss-security/2024/01/20/1","https://devhub.checkmarx.com/cve-details/CVE-2023-50447/","https://duartecsantos.github.io/2024-01-02-CVE-2023-50447/","https://github.com/python-pillow/Pillow/releases","https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html"],"description":"Pillow through 10.1.0 allows PIL.ImageMath.eval Arbitrary Code Execution via the environment parameter, a different vulnerability than CVE-2022-22817 (which was about the expression parameter).","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-50447","epss":0.00408,"percentile":0.6039,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<10.2.0 (python)","vulnerabilityID":"GHSA-3f63-hfp8-52jq"},"fix":{"suggestedVersion":"10.2.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"59.6.0"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"2b9f76b769441aeb","name":"setuptools","version":"59.6.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:59.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@59.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-r9hx-vwmv-q579","dataSource":"https://github.com/advisories/GHSA-r9hx-vwmv-q579","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/setuptools vulnerable to Regular Expression Denial of Service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54322,"date":"2025-06-02"}],"fix":{"versions":["65.5.1"],"state":"fixed"},"advisories":[],"risk":0.2385},"relatedVulnerabilities":[{"id":"CVE-2022-40897","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40897","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/","https://github.com/pypa/setuptools/blob/fe8a98e696241487ba6ac9f91faa38ade939ec5d/setuptools/package_index.py#L200","https://github.com/pypa/setuptools/commit/43a9c9bfa6aa626ec2a22540bea28d2ca77964be","https://github.com/pypa/setuptools/compare/v65.5.0...v65.5.1","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ADES3NLOE5QJKBLGNZNI2RGVOSQXA37R/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/YNA2BAH2ACBZ4TVJZKFLCR7L23BG5C3H/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://pyup.io/vulnerabilities/CVE-2022-40897/52495/","https://security.netapp.com/advisory/ntap-20230214-0001/","https://security.netapp.com/advisory/ntap-20240621-0006/"],"description":"Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40897","epss":0.00318,"percentile":0.54322,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"59.6.0"}},"found":{"versionConstraint":"<65.5.1 (python)","vulnerabilityID":"GHSA-r9hx-vwmv-q579"},"fix":{"suggestedVersion":"65.5.1"}}],"artifact":{"id":"2b9f76b769441aeb","name":"setuptools","version":"59.6.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:59.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@59.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-9j59-75qj-795w","dataSource":"https://github.com/advisories/GHSA-9j59-75qj-795w","namespace":"github:language:python","severity":"High","urls":[],"description":"Path traversal in Pillow","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:H","metrics":{"baseScore":9.1,"exploitabilityScore":3.9,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-24303","epss":0.0023,"percentile":0.4605,"date":"2025-06-02"}],"fix":{"versions":["9.0.1"],"state":"fixed"},"advisories":[],"risk":0.1909},"relatedVulnerabilities":[{"id":"CVE-2022-24303","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-24303","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/python-pillow/Pillow/pull/3450","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/W4ZUXPKEX72O3E5IHBPVY5ZCPMJ4GHHV/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/XR6UP2XONXOVXI4446VY72R63YRO2YTP/","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security","https://security.gentoo.org/glsa/202211-10","https://github.com/python-pillow/Pillow/pull/3450","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/W4ZUXPKEX72O3E5IHBPVY5ZCPMJ4GHHV/","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/XR6UP2XONXOVXI4446VY72R63YRO2YTP/","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security","https://security.gentoo.org/glsa/202211-10"],"description":"Pillow before 9.0.1 allows attackers to delete files because spaces in temporary pathnames are mishandled.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:H","metrics":{"baseScore":9.1,"exploitabilityScore":3.9,"impactScore":5.2},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:P/A:P","metrics":{"baseScore":6.4,"exploitabilityScore":10,"impactScore":5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-24303","epss":0.0023,"percentile":0.4605,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<9.0.1 (python)","vulnerabilityID":"GHSA-9j59-75qj-795w"},"fix":{"suggestedVersion":"9.0.1"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-cx63-2mw6-8hw5","dataSource":"https://github.com/advisories/GHSA-cx63-2mw6-8hw5","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools vulnerable to Command Injection via package URL","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45716,"date":"2025-06-02"}],"fix":{"versions":["70.0.0"],"state":"fixed"},"advisories":[],"risk":0.185005},"relatedVulnerabilities":[{"id":"CVE-2024-6345","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6345","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5","https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"],"description":"A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45716,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"59.6.0"}},"found":{"versionConstraint":"<70.0.0 (python)","vulnerabilityID":"GHSA-cx63-2mw6-8hw5"},"fix":{"suggestedVersion":"70.0.0"}}],"artifact":{"id":"2b9f76b769441aeb","name":"setuptools","version":"59.6.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/setuptools-59.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["UNKNOWN"],"cpes":["cpe:2.3:a:python:setuptools:59.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@59.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-8ghj-p4vj-mr35","dataSource":"https://github.com/advisories/GHSA-8ghj-p4vj-mr35","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow Denial of Service vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-44271","epss":0.00148,"percentile":0.36591,"date":"2025-06-02"}],"fix":{"versions":["10.0.0"],"state":"fixed"},"advisories":[],"risk":0.11099999999999999},"relatedVulnerabilities":[{"id":"CVE-2023-44271","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-44271","namespace":"nvd:cpe","severity":"High","urls":["https://devhub.checkmarx.com/cve-details/CVE-2023-44271/","https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7","https://github.com/python-pillow/Pillow/pull/7244","https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4/","https://devhub.checkmarx.com/cve-details/CVE-2023-44271/","https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7","https://github.com/python-pillow/Pillow/pull/7244","https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4/"],"description":"An issue was discovered in Pillow before 10.0.0. It is a Denial of Service that uncontrollably allocates memory to process a given task, potentially causing a service to crash by having it run out of memory. This occurs for truetype in ImageFont when textlength in an ImageDraw instance operates on a long text argument.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-44271","epss":0.00148,"percentile":0.36591,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":">=0,<10.0.0 (python)","vulnerabilityID":"GHSA-8ghj-p4vj-mr35"},"fix":{"suggestedVersion":"10.0.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-m2vv-5vj5-2hm7","dataSource":"https://github.com/advisories/GHSA-m2vv-5vj5-2hm7","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow vulnerable to Data Amplification attack.","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-45198","epss":0.00123,"percentile":0.32712,"date":"2025-06-02"}],"fix":{"versions":["9.2.0"],"state":"fixed"},"advisories":[],"risk":0.09225},"relatedVulnerabilities":[{"id":"CVE-2022-45198","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-45198","namespace":"nvd:cpe","severity":"High","urls":["https://bugs.gentoo.org/855683","https://cwe.mitre.org/data/definitions/409.html","https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4","https://github.com/python-pillow/Pillow/pull/6402","https://github.com/python-pillow/Pillow/releases/tag/9.2.0","https://security.gentoo.org/glsa/202211-10","https://bugs.gentoo.org/855683","https://cwe.mitre.org/data/definitions/409.html","https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4","https://github.com/python-pillow/Pillow/pull/6402","https://github.com/python-pillow/Pillow/releases/tag/9.2.0","https://security.gentoo.org/glsa/202211-10"],"description":"Pillow before 9.2.0 performs Improper Handling of Highly Compressed GIF Data (Data Amplification).","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-45198","epss":0.00123,"percentile":0.32712,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<9.2.0 (python)","vulnerabilityID":"GHSA-m2vv-5vj5-2hm7"},"fix":{"suggestedVersion":"9.2.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-qwmp-2cf2-g9g6","dataSource":"https://github.com/advisories/GHSA-qwmp-2cf2-g9g6","namespace":"github:language:python","severity":"High","urls":[],"description":"pypa/wheel vulnerable to Regular Expression denial of service (ReDoS)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40898","epss":0.00108,"percentile":0.30277,"date":"2025-06-02"}],"fix":{"versions":["0.38.1"],"state":"fixed"},"advisories":[],"risk":0.08099999999999999},"relatedVulnerabilities":[{"id":"CVE-2022-40898","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40898","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18","https://pypi.org/project/wheel/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L18","https://pypi.org/project/wheel/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"],"description":"An issue discovered in Python Packaging Authority (PyPA) Wheel 0.37.1 and earlier allows remote attackers to cause a denial of service via attacker controlled input to wheel cli.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40898","epss":0.00108,"percentile":0.30277,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"wheel","version":"0.37.1"}},"found":{"versionConstraint":"<0.38.1 (python)","vulnerabilityID":"GHSA-qwmp-2cf2-g9g6"},"fix":{"suggestedVersion":"0.38.1"}}],"artifact":{"id":"13bd44c345c9379d","name":"wheel","version":"0.37.1","type":"python","locations":[{"path":"/lib/python3.6/site-packages/wheel-0.37.1.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/wheel-0.37.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/wheel-0.37.1.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/wheel-0.37.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:daniel_holth_project:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth_project:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth_project:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holthproject:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholth_project:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholth:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholth:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholthproject:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:daniel_holth:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python-wheel:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python_wheel:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:wheel:python-wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:wheel:python_wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:dholth:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:python:wheel:0.37.1:*:*:*:*:*:*:*","cpe:2.3:a:wheel:wheel:0.37.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/wheel@0.37.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-xrcv-f9gm-v42c","dataSource":"https://github.com/advisories/GHSA-xrcv-f9gm-v42c","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Out-of-bounds Read in Pillow","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:L","metrics":{"baseScore":6.5,"exploitabilityScore":3.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-22816","epss":0.00121,"percentile":0.325,"date":"2025-06-02"}],"fix":{"versions":["9.0.0"],"state":"fixed"},"advisories":[],"risk":0.06957499999999998},"relatedVulnerabilities":[{"id":"CVE-2022-22816","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-22816","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/python-pillow/Pillow/blob/c5d9223a8b5e9295d15b5a9b1ef1dae44c8499f3/src/path.c#L331","https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling","https://security.gentoo.org/glsa/202211-10","https://www.debian.org/security/2022/dsa-5053","https://github.com/python-pillow/Pillow/blob/c5d9223a8b5e9295d15b5a9b1ef1dae44c8499f3/src/path.c#L331","https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling","https://security.gentoo.org/glsa/202211-10","https://www.debian.org/security/2022/dsa-5053"],"description":"path_getbbox in path.c in Pillow before 9.0.0 has a buffer over-read during initialization of ImagePath.Path.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:L","metrics":{"baseScore":6.5,"exploitabilityScore":3.9,"impactScore":2.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:P/A:P","metrics":{"baseScore":6.4,"exploitabilityScore":10,"impactScore":5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-22816","epss":0.00121,"percentile":0.325,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<9.0.0 (python)","vulnerabilityID":"GHSA-xrcv-f9gm-v42c"},"fix":{"suggestedVersion":"9.0.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-44wm-f244-xhp3","dataSource":"https://github.com/advisories/GHSA-44wm-f244-xhp3","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow buffer overflow vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":6.7,"exploitabilityScore":0.8,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28219","epss":0.00095,"percentile":0.27871,"date":"2025-06-02"}],"fix":{"versions":["10.3.0"],"state":"fixed"},"advisories":[],"risk":0.06745},"relatedVulnerabilities":[{"id":"CVE-2024-28219","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-28219","namespace":"nvd:cpe","severity":"Medium","urls":["https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/","https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security","https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/","https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security"],"description":"In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.","cvss":[{"source":"cve@mitre.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":6.7,"exploitabilityScore":0.8,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28219","epss":0.00095,"percentile":0.27871,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<10.3.0 (python)","vulnerabilityID":"GHSA-44wm-f244-xhp3"},"fix":{"suggestedVersion":"10.3.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-6p56-wp2h-9hxr","dataSource":"https://github.com/advisories/GHSA-6p56-wp2h-9hxr","namespace":"github:language:python","severity":"Medium","urls":[],"description":"NumPy Buffer Overflow (Disputed)","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.3,"exploitabilityScore":1.7,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-33430","epss":0.00124,"percentile":0.32961,"date":"2025-06-02"}],"fix":{"versions":["1.21"],"state":"fixed"},"advisories":[],"risk":0.06386},"relatedVulnerabilities":[{"id":"CVE-2021-33430","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-33430","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/numpy/numpy/issues/18939","https://github.com/numpy/numpy/issues/18939"],"description":"A Buffer Overflow vulnerability exists in NumPy 1.9.x in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. NOTE: The vendor does not agree this is a vulneraility; In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.3,"exploitabilityScore":1.7,"impactScore":3.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:M/Au:S/C:N/I:N/A:P","metrics":{"baseScore":3.5,"exploitabilityScore":6.9,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-33430","epss":0.00124,"percentile":0.32961,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"numpy","version":"1.19.5"}},"found":{"versionConstraint":">=1.9.0,<1.21 (python)","vulnerabilityID":"GHSA-6p56-wp2h-9hxr"},"fix":{"suggestedVersion":"1.21"}}],"artifact":{"id":"a121bf898d846154","name":"numpy","version":"1.19.5","type":"python","locations":[{"path":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:travis_e__oliphant_et_al__project:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al__project:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al__project:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:numpy:1.19.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/numpy@1.19.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-pw3c-h7wp-cvhx","dataSource":"https://github.com/advisories/GHSA-pw3c-h7wp-cvhx","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Improper Initialization in Pillow","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:L","metrics":{"baseScore":6.5,"exploitabilityScore":3.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-22815","epss":0.00078,"percentile":0.24229,"date":"2025-06-02"}],"fix":{"versions":["9.0.0"],"state":"fixed"},"advisories":[],"risk":0.044849999999999994},"relatedVulnerabilities":[{"id":"CVE-2022-22815","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-22815","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/python-pillow/Pillow/blob/c5d9223a8b5e9295d15b5a9b1ef1dae44c8499f3/src/path.c#L331","https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling","https://security.gentoo.org/glsa/202211-10","https://www.debian.org/security/2022/dsa-5053","https://github.com/python-pillow/Pillow/blob/c5d9223a8b5e9295d15b5a9b1ef1dae44c8499f3/src/path.c#L331","https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html","https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling","https://security.gentoo.org/glsa/202211-10","https://www.debian.org/security/2022/dsa-5053"],"description":"path_getbbox in path.c in Pillow before 9.0.0 improperly initializes ImagePath.Path.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:L","metrics":{"baseScore":6.5,"exploitabilityScore":3.9,"impactScore":2.6},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:P/A:P","metrics":{"baseScore":6.4,"exploitabilityScore":10,"impactScore":5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-22815","epss":0.00078,"percentile":0.24229,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<9.0.0 (python)","vulnerabilityID":"GHSA-pw3c-h7wp-cvhx"},"fix":{"suggestedVersion":"9.0.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-fpfv-jqm9-f5jm","dataSource":"https://github.com/advisories/GHSA-fpfv-jqm9-f5jm","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Incorrect Comparison in NumPy","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-34141","epss":0.00061,"percentile":0.19481,"date":"2025-06-02"}],"fix":{"versions":["1.22"],"state":"fixed"},"advisories":[],"risk":0.031415},"relatedVulnerabilities":[{"id":"CVE-2021-34141","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-34141","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/numpy/numpy/issues/18993","https://www.oracle.com/security-alerts/cpujul2022.html","https://github.com/numpy/numpy/issues/18993","https://www.oracle.com/security-alerts/cpujul2022.html"],"description":"An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. NOTE: the vendor states that this reported code behavior is \"completely harmless.\"","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:N/A:P","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-34141","epss":0.00061,"percentile":0.19481,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"numpy","version":"1.19.5"}},"found":{"versionConstraint":"<1.22 (python)","vulnerabilityID":"GHSA-fpfv-jqm9-f5jm"},"fix":{"suggestedVersion":"1.22"}}],"artifact":{"id":"a121bf898d846154","name":"numpy","version":"1.19.5","type":"python","locations":[{"path":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/numpy-1.19.5.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:travis_e__oliphant_et_al__project:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al__project:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al__project:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:python-numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:python_numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:python:numpy:1.19.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:numpy:1.19.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/numpy@1.19.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-mq26-g339-26xf","dataSource":"https://github.com/advisories/GHSA-mq26-g339-26xf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Command Injection in pip when used with Mercurial","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11732,"date":"2025-06-02"}],"fix":{"versions":["23.3"],"state":"fixed"},"advisories":[],"risk":0.021},"relatedVulnerabilities":[{"id":"CVE-2023-5752","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-5752","namespace":"nvd:cpe","severity":"Low","urls":["https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/","https://github.com/pypa/pip/pull/12306","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/622OZXWG72ISQPLM5Y57YCVIMWHD4C3U/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/65UKKF5LBHEFDCUSPBHUN4IHYX7SRMHH/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FXUVMJM25PUAZRQZBF54OFVKTY3MINPW/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KFC2SPFG5FLCZBYY2K3T5MFW2D22NG6E/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YBSB3SUPQ3VIFYUMHPO3MEQI4BJAXKCZ/","https://mail.python.org/archives/list/security-announce@python.org/thread/F4PL35U6X4VVHZ5ILJU3PWUWN7H7LZXL/"],"description":"When installing a package from a Mercurial VCS URL  (ie \"pip install \nhg+...\") with pip prior to v23.3, the specified Mercurial revision could\n be used to inject arbitrary configuration options to the \"hg clone\" \ncall (ie \"--config\"). Controlling the Mercurial configuration can modify\n how and which repository is installed. This vulnerability does not \naffect users who aren't installing from Mercurial.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@python.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-5752","epss":0.0004,"percentile":0.11732,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pip","version":"21.3.1"}},"found":{"versionConstraint":"<23.3 (python)","vulnerabilityID":"GHSA-mq26-g339-26xf"},"fix":{"suggestedVersion":"23.3"}}],"artifact":{"id":"c2b7e867451975ef","name":"pip","version":"21.3.1","type":"python","locations":[{"path":"/lib/python3.6/site-packages/pip-21.3.1.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/pip-21.3.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/pip-21.3.1.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/pip-21.3.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:pip_developers_project:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers_project:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig_project:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developersproject:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sigproject:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip_developers:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils-sig:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:distutils_sig:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python-pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:python_pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pip:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pip:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pypa:pip:21.3.1:*:*:*:*:*:*:*","cpe:2.3:a:pip:pip:21.3.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pip@21.3.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-4fx9-vc88-q2xc","dataSource":"https://github.com/advisories/GHSA-4fx9-vc88-q2xc","namespace":"github:language:python","severity":"Low","urls":[],"description":"Infinite loop in Pillow","cvss":[],"fix":{"versions":["9.0.0"],"state":"fixed"},"advisories":[],"risk":0},"relatedVulnerabilities":[],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"8.4.0"}},"found":{"versionConstraint":"<9.0.0 (python)","vulnerabilityID":"GHSA-4fx9-vc88-q2xc"},"fix":{"suggestedVersion":"9.0.0"}}],"artifact":{"id":"2b80d73c74be1f4a","name":"pillow","version":"8.4.0","type":"python","locations":[{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","accessPath":"/lib/python3.6/site-packages/Pillow-8.4.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:8.4.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:8.4.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@8.4.0","upstreams":[]}}],"source":{"type":"sbom-file","target":"tablexplore-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/tablexplore-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/tablexplore-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-03T01:32:03Z_1748923962.tar.zst?checksum=sha256%3A02a6127217518a93e0ef8739cb9f2d659109fff0d9cc25f0c13d5ca8b307a118","built":"2025-06-03T04:12:42Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-03T01:32:10Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-03T01:32:15Z","input":"xxh64:bad092583f78d4dd"},"debian":{"captured":"2025-06-03T01:32:22Z","input":"xxh64:b4717cad509f50d8"},"epss":{"captured":"2025-06-03T01:32:43Z","input":"xxh64:2f9660be834e0ce6"},"github":{"captured":"2025-06-03T01:32:21Z","input":"xxh64:cdd94f1952fcf607"},"kev":{"captured":"2025-06-03T01:32:03Z","input":"xxh64:575844c3434cc70b"},"mariner":{"captured":"2025-06-03T01:32:11Z","input":"xxh64:7628f1adb0b9b12b"},"nvd":{"captured":"2025-06-03T01:32:30Z","input":"xxh64:463c4e9a1d401101"},"oracle":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:257df3059d28d823"},"rhel":{"captured":"2025-06-03T01:33:04Z","input":"xxh64:59e7d70bf674afdc"},"sles":{"captured":"2025-06-03T01:32:46Z","input":"xxh64:3cf7f372ea0c981a"},"ubuntu":{"captured":"2025-06-03T01:33:20Z","input":"xxh64:075be8afc080f2be"},"wolfi":{"captured":"2025-06-03T01:32:31Z","input":"xxh64:1aa0b26b004bdd38"}}},"timestamp":"2025-06-04T10:12:36.269785+01:00"}}
